packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.vue

```diff
@@ -52,85 +58,133 @@ const slots = defineSlots<{
-    }
+const onToggle = () => {
+  isExpanded.value = !isExpanded.value;
+  if (!isExpanded.value) {
+    emit("update:searchTerm", "");
+    selectInput.value?.focus();
+  }
+};
 
-    if (!props.multiple) {
-      const newValue = selectedOption === props.modelValue ? undefined : selectedOption;
-      emit("update:modelValue", newValue as typeof props.modelValue);
-      return;
-    }
-    const arrayValues: TValue[] = Array.isArray(props.modelValue) ? props.modelValue : [];
-    const newValues = arrayValues.includes(selectedOption)
-      ? arrayValues.filter((i) => i !== selectedOption)
-      : [...arrayValues, selectedOption];
-    emit("update:modelValue", newValues as typeof props.modelValue);
-  },
-  onActivateFirst: () => (activeOption.value = allKeyboardOptionIds.value.at(0)),
-  onActivateLast: () => (activeOption.value = allKeyboardOptionIds.value.at(-1)),
-  onActivateNext: (currentValue) => {
-    const currentIndex = allKeyboardOptionIds.value.findIndex((i) => i === currentValue);
-    if (currentIndex < allKeyboardOptionIds.value.length - 1) {
-      activeOption.value = allKeyboardOptionIds.value[currentIndex + 1];
-    }
-  },
-  onActivatePrevious: (currentValue) => {
-    const currentIndex = allKeyboardOptionIds.value.findIndex((i) => i === currentValue);
-    if (currentIndex > 0) activeOption.value = allKeyboardOptionIds.value[currentIndex - 1];
-  },
-  onTypeAhead: (label) => {
-    const firstMatch = props.options.find((i) => {
-      return i.label.toLowerCase().trim().startsWith(label.toLowerCase());
-    });
-    if (!firstMatch) return;
-    activeOption.value = firstMatch.value;
-  },
+const onActivateFirst = () => (activeValue.value = allKeyboardOptionIds.value.at(0));
+
+const onActivateLast = () => (activeValue.value = allKeyboardOptionIds.value.at(-1));
+
+const onActivateNext = (currentValue: string) => {
+  const currentIndex = allKeyboardOptionIds.value.findIndex((i) => i === currentValue);
+  if (currentIndex < allKeyboardOptionIds.value.length - 1) {
+    activeValue.value = allKeyboardOptionIds.value[currentIndex + 1];
+  }
+};
+
+const onActivatePrevious = (currentValue: string) => {
+  const currentIndex = allKeyboardOptionIds.value.findIndex((i) => i === currentValue);
+  if (currentIndex > 0) activeValue.value = allKeyboardOptionIds.value[currentIndex - 1];
+};
+
+const onTypeAhead = (label: string) => {
+  const firstMatch = props.options.find((i) => {
+    return i.label.toLowerCase().trim().startsWith(label.toLowerCase());
+  });
+  if (!firstMatch) return;
+  activeValue.value = firstMatch.value;
+};
+
+const onAutocomplete = (inputValue: string) => emit("update:searchTerm", inputValue);
+
+const onSelect = (selectedOption: string) => {
+  if (selectedOption === CHECK_ALL_ID) {
+    checkAll.value?.handleChange(!checkAll.value.state.value.modelValue);
+    return;
+  }
+  const newValue = props.options.find(({ value }) => value === selectedOption);
+  if (!newValue) {
+    return;
+  }
```

Comment:
Like this de-selecting a value does not work, right? Shouldn't we support this?

Reply:
It works, `onSelect` is triggered whenever a option is clicked, even when it's already selected.

Reply:
For a single select, the deselection is not allowed.